### ISSUE TITLE
[v2] showOverlay with same ID

### ISSUE THREAD

**Issue by yusufyildirim** [wix/react-native-navigation#3202]
### Issue Description
Hi, I'm facing with a problem about the overlays and modals. Whenever I try  to show an overlay with existing id, it throw an error. After some research, I found this related issue https://github.com/wix/react-native-navigation/issues/3081. I explained the steps for reproduce this error but I wanna tell you how I solved it too. Here is my temporary solution:

I added a function to Navigation object called setIdFree
`Navigation.setIdFree = componentId => Navigation.store.cleanId(componentId);`

After that, I called this setIdFree function right after whenever I call the dismissOverlay function
`navigation.dismissOverlay('LoadingModal');`
`navigation.setIdFree('LoadingModal')`

With this way, error is gone and everything is working as expected.

I tried with modals too, modals are not throwing error but not showing the second modal too (with same id)

Sum: So, I know showing a modal, overlay or screen with same id is forbidden but even after we dismiss previous one? I think you guys should free the id whenever dismiss function called ( for modals and overlays at least ) or at least you can give us a way to do this manually like I did.

Thank you so much

### Steps to Reproduce / Code Snippets / Screenshots

showOverlay with id: LoadingOverlay
dismissOverlay with id: LoadingOverlay
-everything is fine for now-
after then, showOverlay with id: LoadingOverlay again (nothing appears)
dismissOverlay with id: LoadingOverlay
-error occurs-

![simulator screen shot - iphone 6 - 2018-05-13 at 21 16 05](https://user-images.githubusercontent.com/22980987/39984667-8c85cfa0-5764-11e8-8195-5549f286c434.png)

---
### Environment
* React Native Navigation version: v2@alpha
* React Native version: 0.55.4
* Platform(s) (iOS, Android, or both?): iOS (Didn't test Android)
* Device info (Simulator/Device? OS version? Debug/Release?): Simulator / Debug / iOS 11.3

**Comment by yusufyildirim** [comment]
Lol, I totally understand the situation. It's related with my another issue #3199. There is a class called ComponentWrapper in the library and this class has a componentWillUnmount event like this:

![screen shot 2018-05-14 at 21 25 13](https://user-images.githubusercontent.com/22980987/40015859-52b2e458-57bd-11e8-9508-6daf9b577330.png)

Everything looks right, but the problem is componentWillUnmount event is not firing because overlays and modals are not "really" disappearing when we call dismissModal or dismissOverlay functions. Pretty buggy and performance killer behavior. What I mean is, views are disappearing but their react components are still alive, they are not destroying their components. These components are my overlays which are already "dismissed"

![screen shot 2018-05-15 at 00 13 32](https://user-images.githubusercontent.com/22980987/40024119-f3966ec2-57d5-11e8-89c9-002611c5820d.jpg)

I don't know Obj-C even a bit but I'm an obsessive person and I spent a couple hours for find the problem and yess I think I found it. In the `RNNOverlayManager.m` file there is a function called removeCachedOverlay and it's calling the `[viewController.view removeFromSuperview];` command. As I understand, removeFromSuperview command is just "detaching" the view from it's superview, it's not destroying the views itself. So I set view object to nil right after that and tada it's working like a charm! I'm gonna send a PR about this, if it's the correct approach, I hope you'll merge it or fix it with your way :D
